src/freezing/freezing_main.py

```diff
@@ -35,6 +35,16 @@ def run_freezing(
         constructed_df (pd.DataFrame): As main_snapshot but with records amended
             and added from the freezing files.
     """
+    # return frozen snapshot if config allows
+    run_first_snapshot_of_results = config["global"]["run_first_snapshot_of_results"]
+    frozen_data_staged_path = config["global"]["frozen_data_staged_path"]
+    if not run_first_snapshot_of_results:
+        freezing_logger.info("Loading frozen data...")
+        frozen_snapshot = read_csv(frozen_data_staged_path)
+        freezing_logger.info(
+            "Frozen data successfully read from {frozen_data_staged_path}"
+        )
+        return frozen_snapshot
```

Comment:
Could this be renamed to frozen_data_staged_csv or something similar, so we don't confuse it with the json snapshot

Reply:
done